File: sdkwork-ai-sdk-swift/Sources/API/FileApi.swift
import Foundation

public class FileApi {
    private let client: HttpClient
    
    public init(client: HttpClient) {
        self.client = client
    }

    /// List files
    public func listFiles(params: [String: Any]? = nil) async throws -> OpenAiFileList? {
        let response = try await client.get(ApiPaths.aiPath("/files"), params: params)
        return response as? OpenAiFileList
    }

    /// Upload file
    public func upload(body: UploadFileRequest? = nil, params: [String: Any]? = nil) async throws -> OpenAiFileObject? {
        let response = try await client.post(ApiPaths.aiPath("/files"), body: body, params: params, headers: nil, contentType: "multipart/form-data")
        return response as? OpenAiFileObject
    }

    /// Retrieve file content
    public func retrieveFileContent(file_id: String) async throws -> String? {
        let response = try await client.get(ApiPaths.aiPath("/files/\(file_id)/content"))
        return response as? String
    }

    /// Retrieve file
    public func retrieve(file_id: String) async throws -> OpenAiFileObject? {
        let response = try await client.get(ApiPaths.aiPath("/files/\(file_id)"))
        return response as? OpenAiFileObject
    }

    /// Delete file
    public func deleteFile(file_id: String) async throws -> OpenAiFileDeleteResponse? {
        let response = try await client.delete(ApiPaths.aiPath("/files/\(file_id)"))
        return response as? OpenAiFileDeleteResponse
    }
}

File: sdkwork-ai-sdk-csharp/Api/FileApi.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Ai.Http;
using Ai.Models;

namespace Ai.Api
{
    public class FileApi
    {
        private readonly HttpClient _client;

        public FileApi(HttpClient client)
        {
            _client = client;
        }

        /// <summary>
        /// List files
        /// </summary>
        public async Task<OpenAiFileList?> ListFilesAsync(Dictionary<string, object>? query = null)
        {
            return await _client.GetAsync<OpenAiFileList>(ApiPaths.AiPath("/files"), query);
        }

        /// <summary>
        /// Upload file
        /// </summary>
        public async Task<OpenAiFileObject?> UploadAsync(UploadFileRequest? body = null, Dictionary<string, object>? query = null)
        {
            return await _client.PostAsync<OpenAiFileObject>(ApiPaths.AiPath("/files"), body, query, null, "multipart/form-data");
        }

        /// <summary>
        /// Retrieve file content
        /// </summary>
        public async Task<string?> RetrieveFileContentAsync(string file_id)
        {
            return await _client.GetAsync<string>(ApiPaths.AiPath($"/files/{file_id}/content"));
        }

        /// <summary>
        /// Retrieve file
        /// </summary>
        public async Task<OpenAiFileObject?> RetrieveAsync(string file_id)
        {
            return await _client.GetAsync<OpenAiFileObject>(ApiPaths.AiPath($"/files/{file_id}"));
        }

        /// <summary>
        /// Delete file
        /// </summary>
        public async Task<OpenAiFileDeleteResponse?> DeleteFileAsync(string file_id)
        {
            return await _client.DeleteAsync<OpenAiFileDeleteResponse>(ApiPaths.AiPath($"/files/{file_id}"));
        }
    }
}
